packages/jellyfish-api-core/__tests__/category/icxorderbook/closeOrder.test.ts

```diff
@@ -0,0 +1,350 @@
+      offerTx: makeOfferTxId,
+      closed: true
+    }
+    const HTLCsAfterClaim: Record<string, ICXDFCHTLCInfo | ICXEXTHTLCInfo | ICXClaimDFCHTLCInfo> = await client.icxorderbook.listHTLCs(listHTLCOptionsAfterClaim)
+    expect(Object.keys(HTLCsAfterClaim).length).toBe(4) // extra entry for the warning text returned by the RPC atm.
+    // we have a common field "type", use that to narrow down the record
+    if (HTLCsAfterClaim[claimTxId].type === ICXHTLCType.CLAIM_DFC) {
+      // ICXClaimDFCHTLCInfo cast
+      const ClaimHTLCInfo: ICXClaimDFCHTLCInfo = HTLCsAfterClaim[claimTxId] as ICXClaimDFCHTLCInfo
+      expect(ClaimHTLCInfo.dfchtlcTx).toStrictEqual(DFCHTLCTxId)
+      expect(ClaimHTLCInfo.seed).toStrictEqual('f75a61ad8f7a6e0ab701d5be1f5d4523a9b534571e4e92e0c4610c6a6784ccef')
+    }
+
+    // check HTLC DFCHTLCTxId is in claimed status
+    if (HTLCsAfterClaim[DFCHTLCTxId].type === ICXHTLCType.DFC) {
+      // ICXDFCHTLCInfo cast
+      const DFCHTLCInfo: ICXDFCHTLCInfo = HTLCsAfterClaim[DFCHTLCTxId] as ICXDFCHTLCInfo
+      expect(DFCHTLCInfo.status).toStrictEqual(ICXHTLCStatus.CLAIMED)
+    }
+
+    const accountDFIAfterClaim: Record<string, BigNumber> = await client.call('getaccount', [accountDFI, {}, true], 'bignumber')
+    const accountBTCAfterClaim: Record<string, BigNumber> = await client.call('getaccount', [accountBTC, {}, true], 'bignumber')
+
+    // maker should get incentive + maker deposit and taker should get amount in DFCHTLCTxId HTLC - takerfee
+    expect(accountDFIAfterClaim[idDFI]).toStrictEqual(accountDFIBeforeClaim[idDFI].plus(0.0125))
+    expect(accountBTCAfterClaim[idDFI]).toStrictEqual(accountBTCBeforeClaim[idDFI].plus(10))
+
+    // offer should be close by now
+    const offersForOrder1: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.icxorderbook.listOrders({ orderTx: createOrderTxId })
+    expect(Object.keys(offersForOrder1).length).toBe(1) // extra entry for the warning text returned by the RPC atm.
+
+    // check partial order remaining
+    const ordersAfterClaim: Record<string, ICXOrderInfo | ICXOfferInfo> = await client.icxorderbook.listOrders()
+    const orderInfo: Record<string, ICXOrderInfo> = ordersAfterClaim as Record<string, ICXOrderInfo>
+    expect(orderInfo[createOrderTxId].tokenFrom).toStrictEqual(symbolDFI)
+    expect(orderInfo[createOrderTxId].chainTo).toStrictEqual('BTC')
+    expect(orderInfo[createOrderTxId].ownerAddress).toStrictEqual(accountDFI)
+    expect(orderInfo[createOrderTxId].receivePubkey).toStrictEqual('037f9563f30c609b19fd435a19b8bde7d6db703012ba1aba72e9f42a87366d1941')
+    expect(orderInfo[createOrderTxId].amountFrom).toStrictEqual(new BigNumber(15))
+    expect(orderInfo[createOrderTxId].amountToFill).toStrictEqual(new BigNumber(5))
+    expect(orderInfo[createOrderTxId].orderPrice).toStrictEqual(new BigNumber(0.01))
+    expect(orderInfo[createOrderTxId].amountToFillInToAsset).toStrictEqual(new BigNumber(0.0500000))
```

Comment:
old assertions pattern

Reply:
update in next PR

Reply:
yes, the test is about closing a partially settled order. Hence checking most of the amount details after partial settlement. I could remove one or two like the symbol.